### PR TITLE
fix 1596099. Update permissions with IndexRequest instead of UpdateRe…

### PR DESCRIPTION
**Pull request by jcantrill**
…quest to get overwrite behavior

This PR fixes https://bugzilla.redhat.com/show_bug.cgi?id=1596099 by writing ACL's using IndexRequest instead of UpdateRequest which relies on optimistic concurrency and document merge behavior

### CODE REVIEW

File: src/test/resources/io/fabric8/elasticsearch/plugin/searchguard_roles_acl.yml [hunk 1]
```diff
@@ -13,7 +13,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
+sg_unrecognized_child:
```
Comment (ewolinetz) [inline src/test/resources/io/fabric8/elasticsearch/plugin/searchguard_roles_acl.yml]: how is this test entry used in our tests?

**Comment by ewolinetz** [comment]
question about the test file entry, otherwise LGTM

File: src/test/resources/io/fabric8/elasticsearch/plugin/searchguard_roles_acl.yml (hunk 1 above)
Comment (jcantrill) [reply src/test/resources/io/fabric8/elasticsearch/plugin/searchguard_roles_acl.yml]: Tests a NullPointer I discovered while fixing this issue https://github.com/fabric8io/openshift-elasticsearch-plugin/blob/2.4.4/src/test/java/io/fabric8/elasticsearch/plugin/acl/SearchGuardRoleACLTest.java#L158

**Comment by jcantrill** [comment]
[merge]

**Pull request merged**